### DISCUSSION TITLE
Performing CDK static code analysis before CDK pipeline self-update

### DISCUSSION THREAD

**Discussion by pmadc** [aws/aws-cdk#19902] [Q&A]
What would be the recommended way to add a static code analysis (SCA) step before the cdk pipeline self-update step. We want to run this analysis on the pipeline stack and the application stacks before updating the pipeline. Thank you for your guidance here.

**Comment by indrora** [comment]
The cleanest place to do so would be within the build step (where the CDK runs your application to generate the CloudFormation template).
